library/java/net/openid/appauth/browser/BrowserSelector.java

```diff
@@ -182,8 +182,7 @@ private static boolean hasWarmupService(PackageManager pm, String packageName) {
     private static boolean isFullBrowser(ResolveInfo resolveInfo) {
         // The filter must match ACTION_VIEW, CATEGORY_BROWSEABLE, and at least one scheme,
         if (!resolveInfo.filter.hasAction(Intent.ACTION_VIEW)
-                || !(resolveInfo.filter.hasCategory(Intent.CATEGORY_BROWSABLE)
-                    || (resolveInfo.filter.hasCategory(Intent.CATEGORY_APP_BROWSER)))
+                || !resolveInfo.filter.hasCategory(Intent.CATEGORY_BROWSABLE)
```

Comment:
Reverted category check to code before the https://github.com/openid/AppAuth-Android/pull/558/ thus matching the official documentation.

Reply:
Here "resolveInfo.filter.hasCategory(Intent.CATEGORY_APP_BROWSER)" Fixed for example Opera support.

Reply:
It may be true, but it's an integration issue with their application and imho we shouldn't work around bugs in 3rd party browsers because of it.